api/rpc/project/project.go

```diff
@@ -1,40 +1,105 @@
 package project
 
-// import (
-//
-// 	"encoding/json"
-// 	"golang.org/x/net/context"
-// 	"log"
-// )
-//
-// const (
-// 	keySpace = "/amp/project"
-// )
-//
-// // Service is used to implement ProjectServer
-// type Service struct {
-// }
-//
-// // CreateProject implements ProjectServer
-// func (s *Service) Create(ctx context.Context, in *CreateRequest) (*CreateReply, error) {
-// 	// Storing the project
-// 	etc.Put(keySpace, in)
-//
-// 	// Iterate through all entries
-// 	all, err := etc.All(keySpace)
-// 	if err != nil {
-// 		return nil, err
-// 	}
-// 	for _, node := range all {
-// 		// Deserialize node value into a CreateRequest (could also be just a map[string]interface{}).
-// 		var cr project.CreateRequest
-// 		err := json.Unmarshal([]byte(node.Value), &cr)
-// 		cr.Id = node.Key
-// 		if err != nil {
-// 			return nil, err
-// 		}
-// 		log.Printf("%+v\n", cr)
-// 	}
-//
-// 	return &CreateReply{Message: "Hello " + in.Name}, nil
-// }
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/appcelerator/amp/data/storage"
+	"github.com/golang/protobuf/proto"
+)
+
+const (
+	defTimeout = 5 * time.Second
+	prefix     = "project"
+)
+
+// Proj structure to implement ProjServer interface
+type Proj struct {
+	Store storage.Interface
+}
+
+// Create adds a new entry to the k,v data store
+func (p *Proj) Create(ctx context.Context, req *Project) *ProjectReply {
+	ctx, cancel := context.WithTimeout(context.Background(), defTimeout)
```

Comment:
I like the timeout concept, however, since we delegate the processing to a third party (in this case etcd), i would let this underlying library time out by itself and return the error as is. That would be a different story if we performed a long running operation ourselves.